src/eredis_sub_client.erl

```diff
@@ -379,42 +354,77 @@ queue_or_send(Msg, State) ->
 %% synchronous and if Redis returns something we don't expect, we
 %% crash. Returns {ok, State} or {error, Reason}.
 connect(#state{host = Host, port = Port, socket_options = SocketOptions,
+               transport = Transport,
                connect_timeout = ConnectTimeout, tls_options = TlsOptions,
                auth_cmd = AuthCmd, database = Db} = State) ->
     case eredis_client:connect(Host, Port, SocketOptions, TlsOptions,
                                ConnectTimeout, AuthCmd, Db) of
         {ok, Socket} ->
-            {ok, State#state{socket = Socket}};
+            %% Re-subscribe to channels. Channels are stored in reverse order in
+            %% state.
+            ok = send_subscribe_command(Transport, Socket, "SUBSCRIBE",
+                                        lists:reverse(State#state.channels)),
+            ok = send_subscribe_command(Transport, Socket, "PSUBSCRIBE",
+                                        lists:reverse(State#state.pchannels)),
+            ok = setopts(Socket, Transport, [{active, once}]),
+
+            %% Notify application that connection is ready.
+            send_to_controller({eredis_connected, self()}, State),
+
+            %% In case the connection terminates immediately (this happens with
+            %% an expired certificate with TLS 1.3) schedule a reconnect already
+            %% so that we don't try to reconnect if an error is received before
+            %% reconnect_sleep milliseconds has elapsed.
+            {ok, schedule_reconnect(unknown, State#state{socket = Socket})};
```

Comment:
Isn't this already scheduled in `eredis_client:connect`?

Reply:
No, only in connect/1, not in connect/7.